src/extensions/metadata.rs

```diff
@@ -0,0 +1,181 @@
+//! Adds support for the IMAP METADATA command specificed in [RFC5464](https://tools.ietf.org/html/rfc5464).
+
+use std::fmt::Debug;
+
+use async_std::io::{Read, Write};
+use async_std::prelude::*;
+use async_std::stream::Stream;
+use async_std::sync;
+use imap_proto::types::{MailboxDatum, Metadata};
+use imap_proto::{RequestId, Response};
+
+use crate::client::{validate_str, Session};
+use crate::codec::ResponseData;
+use crate::error::Result;
+use crate::parse::handle_unilateral;
+use crate::types::UnsolicitedResponse;
+
+trait CmdListItemFormat {
+    fn format_as_cmd_list_item(&self) -> String;
+}
+
+impl CmdListItemFormat for Metadata {
+    fn format_as_cmd_list_item(&self) -> String {
+        format!(
+            "{} {}",
+            validate_str(self.entry.as_str()).unwrap(),
+            self.value
+                .as_ref()
+                .map(|v| validate_str(v.as_str()).unwrap())
+                .unwrap_or("NIL".to_string())
+        )
+    }
+}
+
+/// Represents variants of DEPTH parameters for GETMETADATA command
+#[derive(Debug, Copy, Clone)]
+pub enum MetadataDepth {
+    /// Depth 0 for get metadata
+    Zero,
+    /// Depth 1 for get metadata
+    One,
+    /// Depth infinity for get metadata
+    Inf,
```

Comment:
the doc strings do not say what you get if you specify Zero, One or Inf

Reply:
Description and reference to RFC 5464 are added.